File: bin/trades.py
#!/usr/bin/env python

import asyncio
import websockets
import json

# To subscribe to this channel:
msg = \
    {"jsonrpc": "2.0",
     "method": "public/subscribe",
     "id": 42,
     "params": {
        "channels": ["trades.future.BTC.100ms"]}
    }

async def call_api(msg):
   async with websockets.connect('wss://test.deribit.com/ws/api/v2') as websocket:
       await websocket.send(msg)
       while websocket.open:
           response = await websocket.recv()
           # do something with the notifications...
           print(response)

asyncio.get_event_loop().run_until_complete(call_api(json.dumps(msg)))

File: bin/trades.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'rubygems'
require 'websocket-client-simple'
require 'json'

ws = WebSocket::Client::Simple.connect 'wss://www.deribit.com/ws/api/v2'

ws.on :message do |msg|
  p "msg: #{msg.data}"
end

ws.on :open do |e|
  p "open: #{e}"
end

ws.on :close do |e|
  p "close: #{e}"
  exit 1
end

ws.on :error do |e|
  p "error: #{e}"
end

sleep 3

puts 'Subscribing...'
payload = {
  'jsonrpc' => '2.0',
  'method' => 'public/subscribe',
  'id' => 66,
  'params' => {
    'channels' => ['trades.BTC-PERPETUAL.raw']
  }
}
ws.send payload.to_json.to_s

while ws.open? do
  sleep 1
  puts 'Running...'
end
